### PR TITLE
feat: add verified launchd bootstrap primitive

### PR DESCRIPTION
**Pull request by EtanHey**
## Summary
- Add `brainlayer.launchd_primitive` with reusable launchd target, loaded-state probe, raising verifier, and `install_and_verify_launchagent(label, plist_path)` positive post-condition.
- Wire doctor launchd checks through the shared verifier so required-but-unloaded labels fail with label, target, and reason details.
- Reuse the primitive from health-check and CLI bootstrap paths while keeping launchd imports lazy on the CLI fast path.

## Test plan
- RED: `pytest tests/test_doctor.py` failed before implementation with missing `brainlayer.launchd_primitive` and missing doctor target details.
- GREEN: `ruff check src/brainlayer/launchd_primitive.py src/brainlayer/doctor.py src/brainlayer/health_check.py src/brainlayer/cli/__init__.py tests/test_doctor.py`
- GREEN: `pytest tests/test_doctor.py` (10 passed)
- GREEN: `pytest tests/test_doctor.py tests/test_stability_health_check.py tests/test_cli_launchd_mode_a.py` (27 passed)
- GREEN: `.venv/bin/python -m pytest tests/test_cli_direct_sqlite.py::test_cli_stats_p95_under_200ms_with_direct_readonly_store`
- GREEN pre-push gate: pytest unit suite (3077 passed, 9 skipped, 61 deselected, 1 xfailed), MCP tool registration (3 passed), isolated eval/hook routing (40 passed), bun test (1 passed), FTS5 regression shell passed.

## Notes
- Local `python3.13` full `pytest` collection is blocked by an environment mismatch (`numba` rejects installed NumPy 2.4); the repo pre-push hook created/used the Python 3.12 venv and passed the full gate.
- Local CodeRabbit pre-commit found one MAJOR issue in malformed command-runner handling; fixed with a regression test. A second local CodeRabbit run was rate-limited, so PR-level review is requested below.

---

> [!NOTE]
> **Medium Risk**
> Changes launchd bootstrap and verification for core BrainLayer LaunchAgents (watch, drain, enrichment); failures are now explicit, which is safer but alters resume/reconcile exit behavior on partial failure.
> 
> **Overview**
> Introduces **`brainlayer.launchd_primitive`** as the shared place for LaunchAgent install and loaded-state checks, replacing duplicated `launchctl` sequences in the CLI, doctor, and health-check.
> 
> **Bootstrap and verify:** `install_and_verify_launchagent` runs enable/bootout/bootstrap and **requires** a positive loaded post-condition via `launchctl print`. Structured errors (`LaunchdVerificationError`, `LaunchdLabelNotLoadedError`, `LaunchdCommandError`) carry label, target, reason, and command output for reporting.
> 
> **Doctor** uses `verify_launchd_label_loaded` so missing required labels become **fatal** issues with rich `issue_details` instead of loose bool handling.
> 
> **CLI** `resume` and `reconcile-launchd` go through `_bootstrap_labels_or_exit`: every label is attempted even when one fails; verification failures exit **1** with stderr messages, and **`resume` keeps the pause sentinel** if bootstrap does not verify.
> 
> **Health-check** heal/bootstrap paths delegate to the same primitive; failed verify surfaces as `bootstrap_failed:` rather than silent success.
> 
> Tests cover primitive edge cases (already bootstrapped, malformed runner, no unscoped `launchctl list` fallback) and CLI multi-label failure behavior.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8d38e26ee866a87a2e5cfe7ccd7cd7e60076a419. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Add verified launchd bootstrap primitive with structured error handling
> - Introduces [launchd_primitive.py](https://github.com/EtanHey/brainlayer/pull/547/files#diff-b103eca0b1a87ad4f67e0d1b3307289c28ad7cb92de56786efd4aee3e8a6cd39) with `install_and_verify_launchagent`, `is_launchd_label_loaded`, and `verify_launchd_label_loaded` as centralized primitives for bootstrapping and verifying launchd agents.
> - Adds `LaunchdVerificationError`, `LaunchdLabelNotLoadedError`, and `LaunchdCommandError` exception classes that carry structured context (label, target, reason, command outputs) via `issue_details()`.
> - Refactors `cli._bootstrap_label`, `health_check._bootstrap_if_absent`, and `doctor._check_launchd` to delegate to these primitives, replacing duplicated inline enable/bootout/bootstrap/print sequences.
> - Adds `cli._bootstrap_labels_or_exit` to accumulate failures across multiple labels and exit non-zero if any verification fails, so `resume` and `reconcile-launchd` now attempt all labels before reporting.
> - Behavioral Change: `_run_launchctl` now returns `CompletedProcess` instead of an integer return code, affecting any callers that expected an `int`.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 8d38e26.</sup>
> 
>

### CODE REVIEW

File: src/brainlayer/cli/__init__.py
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/brainlayer/cli/__init__.py:1]: Fixed in 8d38e26: _run_launchctl now returns the CompletedProcess so stderr/stdout are preserved for launchd state parsing.

File: src/brainlayer/launchd_primitive.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        command_runner=command_runner,
+    )
+    if bootout_existing:
+        command_runner(["launchctl", "bootout", target])
+    _run_required_launchctl(
+        ["launchctl", "bootstrap", domain, str(resolved_plist_path)],
+        label=label,
+        plist_path=resolved_plist_path,
+        command_runner=command_runner,
+    )
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/brainlayer/launchd_primitive.py]: Fixed in 8d38e26: bootstrap now verifies loaded state after a nonzero bootstrap and accepts an already-loaded service before raising. Added the regression test.

File: src/brainlayer/cli/__init__.py [hunk 2]
```diff
@@ -77,11 +79,9 @@ def _plist_for_launchd_label(
...
 
 def _bootstrap_label(label: str, plist_path: Path) -> None:
-    target = _launchd_target(label)
-    _run_launchctl(["launchctl", "enable", target])
-    _run_launchctl(["launchctl", "bootout", target])
-    _run_launchctl(["launchctl", "bootstrap", f"gui/{os.getuid()}", str(plist_path.expanduser())])
-    _run_launchctl(["launchctl", "print", target])
+    from ..launchd_primitive import install_and_verify_launchagent
+
+    install_and_verify_launchagent(label, plist_path, command_runner=_run_launchctl)
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/brainlayer/cli/__init__.py:87]: Fixed in 8d38e26: resume/reconcile now attempt all labels, aggregate launchd verification failures, fail loudly, and preserve the pause sentinel on failed resume. Added regression tests.

File: src/brainlayer/launchd_primitive.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        return True
+
+    target = launchd_target(label)
+    print_result = command_runner(["launchctl", "print", target])
+    print_state = _loaded_state_from_result(print_result)
+    if print_state is not None:
+        return print_state
+
+    list_result = command_runner(["launchctl", "list", label])
+    return _loaded_state_from_result(list_result)
```
[the remark this answers is not in the document]
Comment (EtanHey) [reply src/brainlayer/launchd_primitive.py]: Fixed in 8d38e26: removed the unscoped launchctl list fallback and added a regression test so indeterminate print output stays unknown instead of falsely loaded.
